build/storage/scripts/disk_infrastructure.py

```diff
@@ -0,0 +1,207 @@
+            "params": {
+                "nqn": nqn,
+                "listen_address": {
+                    "trtype": "TCP",
+                    "adrfam": "IPv4",
+                    "traddr": ip_addr,
+                    "trsvcid": port_to_expose,
+                },
+            },
+        },
+    ]
+    for request in requests:
+        send_rpc_request(
+            request=request,
+            addr=ip_addr,
+            port=storage_target_port,
+        )
+
+
+def create_ramdrive_and_attach_as_ns_to_subsystem(
+    ip_addr: str,
+    ramdrive_name: str,
+    ramdrive_size_in_mb: int,
+    nqn: str,
+    storage_target_port: int,
+) -> str:
+    requests = [
+        {
+            "method": "bdev_malloc_create",
+            "params": {
+                "name": ramdrive_name,
+                "num_blocks": ramdrive_size_in_mb * 1024 * 1024 // 512,
+                "block_size": 512,
+            },
+        },
+        {
+            "method": "nvmf_subsystem_add_ns",
+            "params": {"nqn": nqn, "namespace": {"bdev_name": ramdrive_name}},
+        },
+        {"method": "bdev_get_bdevs", "params": {"name": ramdrive_name}},
+    ]
+    for request in requests:
+        response = send_rpc_request(
+            request=request,
+            addr=ip_addr,
+            port=storage_target_port,
+        )
+    device_uuid = response[0]["uuid"]
+    return device_uuid
+
+
+def uuid2base64(device_uuid: str) -> str:
+    return base64.b64encode(uuid.UUID(device_uuid).bytes).decode()
+
+
+def create_virtio_blk_without_disk_check(
+    ipu_storage_container_ip: str,
+    volume_id: str,
+    physical_id: str,
+    virtual_id: str,
+    hostnqn: str,
+    traddr: str,
+    trsvcid: str,
+    sma_port: int,
+) -> str:
+    request = {
+        "method": "CreateDevice",
+        "params": {
+            "volume": {
+                "volume_id": uuid2base64(volume_id),
+                "nvmf": {
+                    "hostnqn": hostnqn,
+                    "discovery": {
+                        "discovery_endpoints": [
+                            {"trtype": "tcp", "traddr": traddr, "trsvcid": trsvcid}
+                        ]
+                    },
+                },
+            },
+            "virtio_blk": {"physical_id": physical_id, "virtual_id": virtual_id},
+        },
+    }
+    response = send_sma_request(
+        request=request,
+        addr=ipu_storage_container_ip,
+        port=sma_port,
+    )
+    device_handle = response["handle"]
+    return device_handle
+
+
+def delete_virtio_blk(
+    ipu_storage_container_ip: str, device_handle: str, sma_port: int
+) -> int:
+    request = {"method": "DeleteDevice", "params": {"handle": device_handle}}
+    try:
+        send_sma_request(request, ipu_storage_container_ip, sma_port)
+    except Exception as ex:
+        logging.error(ex)
+        return 1
+    return 0
+
+
+def wait_for_virtio_blk_in_os(timeout: float) -> None:
```

Comment:
this is a simplified impl, still we can consider providing there some default wait time which is suitable